build refactoring

rzetelskik opened the pull request.

zimnx commented on Makefile:
```diff
@@ -48,13 +97,12 @@ fmt:
-docker-push:
-	docker push ${RECOMMENDER_IMG}
-	docker push ${UPDATER_IMG}
+images:
```
you need build dependency here, because you need binaries to be built.

zimnx reviewed (commented):
comments inline, please improve commit messages to contain summary of changes

rzetelskik committed:
Local build, common Dockerfile and updated github workflow 

Refactored build to match Scylla Operator.
- Build is now carried out locally.
- Common Docker build stage for all images.
- Verifying dependencies.
- Updated github workflow.
- Formatted updated_test.

rzetelskik committed:
Add vendors

Added vendors' directory.

zimnx reviewed (approved):
LGTM, thanks
